### ISSUE TITLE
Extension host process runs at version N+1

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#58687]
We have observed, sometimes, updates are applied to disk while VS Code is running. It is therefore possible to end up in a situation where VS Code is launched at version N, with a renderer at version N and an extension host at version N, and then the extension host crashes or needs to be restarted and then an extension host at version N+1 is launched since the update was applied on disk.

Since it is very difficult to prevent the updater from touching the disk while VS Code is running, we should implement an extra protection mechanism into the extension host protocol. (fyi @Tyriar @joaomoreno)

**Comment by Tyriar** [comment]
Related: https://github.com/Microsoft/vscode/issues/5908

**Comment by alexandrudima** [comment]
Yes, that's why I created a new issue... because the old bugs keep getting closed ... 😄.
